src/shared/mkfs-util.c

```diff
@@ -225,6 +239,13 @@ int make_filesystem(
                                       "-U", vol_id,
                                       node, NULL);
 
+                else if (streq(fstype, "squashfs"))
+
+                        (void) execlp(mkfs, mkfs,
+                                      root, node,
+                                      "-quiet",
+                                      "-noappend",
+                                      NULL);
```

Comment:
btw, did you see the `-p <filename> d 0755 0 0` switch? That would allow us to fold MakeDirectories= into the command line

Reply:
That would work for `MakeDirectories`, but we'd still need a temporary source tree for source trees from CopyFiles= that are not targeted at "/", so since we need the logic for merging regardless, I say we stick with that for now. If we really end up running into performance issues, we can revisit this later.